hyperon_das_atomdb/adapters/redis_mongo_db.py

```diff
@@ -108,36 +108,28 @@ def create(
             key, value = next(iter(kwargs.items()))  # only one key-value pair
             conditionals = {key: {"$eq": value}}
 
-        if fields:
-            index_id = f"{atom_type}_{self.generate_index_id(','.join(fields), conditionals)}" + (
-                f"_{index_type.value}" if index_type else ""
-            )
-        else:
-            index_id = f"{atom_type}_{self.generate_index_id(field, conditionals)}" + (
-                f'_{index_type.value}' if index_type else ""
-            )
+        index_id = f"{atom_type}_{self.generate_index_id(','.join(fields), conditionals)}" + (
+            f"_{index_type.value}" if index_type else ""
+        )
         index_type: MongoIndexType = index_type or (
-            MongoIndexType.COMPOUND if fields is not None else MongoIndexType.FIELD
+            MongoIndexType.COMPOUND if len(fields) > 1 else MongoIndexType.FIELD
         )
         index_props = {
             'index_type': index_type,
             'conditionals': conditionals,
             'index_name': index_id,
-            'fields': fields or [field],
+            'fields': fields,
         }
+
         index_conditionals = {"name": index_id}
 
         if conditionals:
             index_conditionals["partialFilterExpression"] = index_props['conditionals']
 
         if index_type == MongoIndexType.TEXT:
-            index_list = [(f, 'text') for f in fields] if fields is not None else [(field, 'text')]
+            index_list = [(f, 'text') for f in fields]
         else:
-            index_list = (
-                [(f, ASCENDING) for f in fields]
-                if fields is not None
-                else [(field, ASCENDING)]  # store the index in ascending order
-            )
+            index_list = [(f, ASCENDING) for f in fields]  # store the index in ascending order
```

Comment:
What happens here if `fields` as passed in as an empty list?

Reply:
Added verification